test/config_cat/config_fetcher_test.exs

```diff
@@ -164,6 +190,110 @@ defmodule ConfigCat.ConfigFetcherTest do
+
+      eu_url = eu_config_url(sdk_key)
+
+      redirect_path =
+        Map.get(config, Constants.preferences())
+        |> Map.get(Constants.preferences_base_url())
+
+      redirect_url = config_url(redirect_path, sdk_key)
+
+      # First call: call global-eu, then redirect
+      MockAPI
+      |> expect(:get, 1, fn ^eu_url, _headers, [] ->
+        {:ok, %Response{status_code: 200, body: config}}
+      end)
+      |> expect(:get, 1, fn ^redirect_url, _headers, [] ->
+        {:ok, %Response{status_code: 200, body: config}}
+      end)
+
+      assert {:ok, ^config} = ConfigFetcher.fetch(fetcher)
+
+      # Second call: only the redirected URL
+      MockAPI
+      |> expect(:get, 0, fn ^eu_url, _headers, [] ->
+        {:ok, %Response{status_code: 200, body: config}}
+      end)
+      |> expect(:get, 1, fn ^redirect_url, _headers, [] ->
+        {:ok, %Response{status_code: 200, body: config}}
+      end)
+
+      assert {:ok, ^config} = ConfigFetcher.fetch(fetcher)
+    end
+
+    test "redirection with custom endpoint",
+         %{config_with_redirect: config, sdk_key: sdk_key} = context do
+      base_url = "https://custom.service.net"
+
+      {:ok, fetcher} =
+        start_fetcher(context, data_governance: DataGovernance.eu_only(), base_url: base_url)
+
+      custom_url = config_url(base_url, sdk_key)
+      eu_url = eu_config_url(sdk_key)
+
+      redirect_path =
+        Map.get(config, Constants.preferences())
+        |> Map.get(Constants.preferences_base_url())
+
+      redirect_url = config_url(redirect_path, sdk_key)
+
+      # First call: call custom, no call should be made to the eu endpoint or the redirect one
+      MockAPI
+      |> expect(:get, 1, fn ^custom_url, _headers, [] ->
+        {:ok, %Response{status_code: 200, body: config}}
+      end)
+      |> expect(:get, 0, fn ^eu_url, _headers, [] ->
+        {:ok, %Response{status_code: 200, body: config}}
+      end)
+      |> expect(:get, 0, fn ^redirect_url, _headers, [] ->
+        {:ok, %Response{status_code: 200, body: config}}
+      end)
+
+      assert {:ok, ^config} = ConfigFetcher.fetch(fetcher)
+    end
+
+    test "redirection with force redirect",
+         %{config_with_force_redirect: config, sdk_key: sdk_key} = context do
+      {:ok, fetcher} = start_fetcher(context, data_governance: DataGovernance.eu_only())
+
+      eu_url = eu_config_url(sdk_key)
+
+      redirect_path =
+        Map.get(config, Constants.preferences())
+        |> Map.get(Constants.preferences_base_url())
+
+      redirect_url = config_url(redirect_path, sdk_key)
+
+      # First call: call global, calls should be made normally
+      MockAPI
+      |> expect(:get, 1, fn ^eu_url, _headers, [] ->
+        {:ok, %Response{status_code: 200, body: config}}
+      end)
+      |> expect(:get, 1, fn ^redirect_url, _headers, [] ->
+        {:ok, %Response{status_code: 200, body: config}}
+      end)
+
+      assert {:ok, ^config} = ConfigFetcher.fetch(fetcher)
+    end
+  end
+
```

Comment:
I miss some cases here.
This is a really important and crucial part of the SDKs. Please implement all of the tests from https://github.com/configcat/ruby-sdk/blob/master/spec/datagovernance_spec.rb